deployment/services/redpanda.ts

```diff
@@ -0,0 +1,198 @@
+import * as k8s from '@pulumi/kubernetes';
+import * as pulumi from '@pulumi/pulumi';
+
+export type Redpanda = ReturnType<typeof deployRedpanda>;
+
+export function deployRedpanda() {
+  const redpandaConfig = new pulumi.Config('redpanda');
+  const replicas = redpandaConfig.getNumber('replicas') || 3;
+  const storageSize = redpandaConfig.get('storageSize') || '20Gi';
+  const memoryLimit = redpandaConfig.get('memoryLimit') || '1Gi';
+  const cpuLimit = redpandaConfig.get('cpuLimit') || '1000m';
+
+  const labels = { app: 'redpanda' };
+
+  // StatefulSet for Redpanda
+  const statefulSet = new k8s.apps.v1.StatefulSet('redpanda', {
+    metadata: {
+      name: 'redpanda',
+    },
+    spec: {
+      serviceName: 'redpanda',
+      replicas,
+      selector: {
+        matchLabels: labels,
+      },
+      template: {
+        metadata: {
+          labels,
+        },
+        spec: {
+          containers: [
+            {
+              name: 'redpanda',
+              image: 'redpandadata/redpanda:v25.3.1',
+              imagePullPolicy: 'Always',
+              resources: {
+                limits: {
+                  cpu: cpuLimit,
+                  memory: memoryLimit,
+                },
+              },
+              args: [
+                'redpanda',
+                'start',
+                '--overprovisioned',
+                '--smp',
+                '1',
+                '--memory',
+                memoryLimit,
+                '--kafka-addr',
+                'PLAINTEXT://0.0.0.0:9092',
+                '--advertise-kafka-addr',
+                pulumi.interpolate`PLAINTEXT://\${HOSTNAME}.redpanda.default.svc.cluster.local:9092`,
+              ],
+              ports: [
+                { containerPort: 9092, name: 'kafka' },
+                { containerPort: 8082, name: 'http' },
+                { containerPort: 33145, name: 'rpc' },
+                { containerPort: 9644, name: 'admin' },
+              ],
+              volumeMounts: [
+                {
+                  name: 'datadir',
+                  mountPath: '/var/lib/redpanda/data',
+                },
+              ],
+              livenessProbe: {
+                httpGet: {
+                  path: '/v1/status/ready',
```

Comment:
for most of our services we use `_health`. I'm not sure if we want to keep this consistent or if we don't mind...